pkg/proxy/healthcheck/worker.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// LoadBalancer Healthcheck responder library for kubernetes network proxies
+package healthcheck // import "k8s.io/kubernetes/pkg/proxy/healthcheck"
+import (
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+var once sync.Once
+var healthchecker *proxyHC
+
+func init() {
+	run()
```

Comment:
i dislike this, can we please find a better to kick off run? so many confusion things will happen if you call it from init. I'm guessing you realize this is run when the package is initialized (https://golang.org/ref/spec#Package_initialization), which is very rarely a good idea (look at how python turned out).